source/Lexer.cpp

```diff
@@ -656,6 +658,10 @@ TokenKind Lexer::lexDirective(Token::Info* info) {
     }
 
     info->extra = getDirectiveKind(lexeme().subString(1));
+    if (!onNewLine && std::get<SyntaxKind>(info->extra) != SyntaxKind::MacroUsage) {
+        // All directives other than a macro usage must be on their own line
+        addError(DiagCode::DirectiveNotFirstOnLine, currentOffset() - 1);
```

Comment:
currentOffset() here points to wherever the lexer is looking, so it's at the end of the directive. We probably would like to point the error at the beginning of the directive token (so just store the result of currentOffset() before we advance the pointer above in scanIdentifier()).